app/src/lib/git/stash.ts

```diff
@@ -21,43 +21,23 @@ export const DesktopStashEntryMarker = '!!GitHub_Desktop'
  */
 const desktopStashEntryMessageRe = /!!GitHub_Desktop<(.+)>$/
 
-/**
- * Gets the number of stash entries for the given repository
- */
-export async function getStashSize(repository: Repository): Promise<number> {
-  const delimiter = '1F'
-  const format = ['%gd', '%H', '%gs'].join(`%x${delimiter}`)
-
-  const result = await git(
-    ['log', '-g', '-z', `--pretty=${format}`, 'refs/stash'],
-    repository.path,
-    'getStashSize',
-    {
-      successExitCodes: new Set([0, 128]),
-    }
-  )
-
-  // There's no refs/stashes reflog in the repository or it's not
-  // even a repository. In either case we don't care
-  if (result.exitCode === 128) {
-    return 0
-  }
-
-  const stashEntries = result.stdout
-    .split('\0')
-    .filter(s => s !== '')
-    .filter(s => !desktopStashEntryMessageRe.test(s))
-
-  return stashEntries.length
-}
-
 /**
  * Get the list of stash entries created by Desktop in the current repository
- * using the default ordering of `git stash list` (i.e., LIFO ordering).
+ * using the default ordering of refs (wich is LIFO ordering),
+ * as well as the total amount of stash entries.
  */
-export async function getDesktopStashEntries(
+export async function getStashes(
   repository: Repository
-): Promise<ReadonlyArray<IStashEntry>> {
+): Promise<{
+  /** The stash entries created by Desktop */
```

Comment:
:art: given the presence of code comments here, this feels like it could be extracted to it's own internal type after we land this PR to keep things more organized...